tests/functional/UseCase/Bucket/GrantRevokeBucketAccessToReadOnlyRoleTest.php

```diff
@@ -122,5 +125,31 @@ public function testWSCanAccessExternalBucketAfterGrantingToRO(): void
             )
         );
         $this->assertEquals([['ID' => '1']], $data);
+
+        // REVOKE SELECT on ext bucket from RO
+        $handler = new RevokeBucketAccessFromReadOnlyRoleHandler($this->sessionManager);
+        $command = (new RevokeBucketAccessFromReadOnlyRoleCommand())
+            ->setBucketObjectName($externalBucketDatabaseName)
+            ->setProjectReadOnlyRoleName($this->mainProjectResponse->getProjectReadOnlyRoleName());
+
+        $handler(
+            $this->mainProjectCredentials,
+            $command,
+            []
+        );
+
+        // check that the Project2 cannot access the table
+        try {
+            $wsConnection->fetchAllAssociative(
+                sprintf(
+                    'SELECT * FROM %s.%s',
+                    TeradataQuote::quoteSingleIdentifier($externalBucketDatabaseName),
+                    TeradataQuote::quoteSingleIdentifier('TESTTABLE')
+                )
+            );
+            $this->fail('Should fail. Bucket is no longer registered as external');
+        } catch (Throwable $e) {
+            $this->assertStringContainsString('The user does not have SELECT access to', $e->getMessage());
```

Comment:
to je ta exception co hledas tady ? https://github.com/keboola/storage-api-php-client/pull/997#issuecomment-1424354103

Reply:
jo, to bude nejspis ona